### PR TITLE
update GHCN converter including height to stationElevation

### LINKED ISSUE #1326: GHCN snow depth decoder height to stationElevation
## Current behavior (describe the bug)

the GHCN snow depth decoder should record the height of station as stationElevation

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

the metaData keyword `height` should be changed to `stationElevation`

there are likely test files which will need to be regenerated with this new keyword

there are a lot of small changes coming up a refactor for efficiency is also needed not sure if tackling this yet
but some other changes

produce real epoch
modify for date can be something other than 18Z
fix missing values to standard ones

## Issue(s) addressed

Resolves #1326 
also address #1178

### CODE REVIEW

File: src/land/ghcn_snod2ioda.py [hunk 1]
```diff
@@ -83,34 +92,41 @@ def assignValue(colrowValue, df400):
...
 
         df30 = pd.concat(df30_list, ignore_index=True)
         df30 = df30[df30["ELEMENT"] == "SNWD"]
         df30["DATETIME"] = df30.apply(lambda row: parse(str(row["DATETIME"])).date(), axis=1)
         # select data with Start date
         startdate = self.date
-        df30 = df30[df30["DATETIME"] == parse(startdate).date()]
+        valid_date = datetime.strptime(startdate, "%Y%m%d%H")
+        select_hour = int(valid_date.strftime('%H'))
+        if select_hour < 12:
```
Comment (BenjaminRuston) [inline src/land/ghcn_snod2ioda.py:106]:
```suggestion
```

File: src/land/ghcn_snod2ioda.py [hunk 2]
```diff
@@ -83,34 +92,41 @@ def assignValue(colrowValue, df400):
...
+            valid_date -= timedelta(days=1)
```
Comment (BenjaminRuston) [inline src/land/ghcn_snod2ioda.py:107]:
```suggestion
```

File: src/land/ghcn_snod2ioda.py [hunk 3]
```diff
@@ -150,22 +165,24 @@ def assignValue(colrowValue, df400):
...
-                errs[i] = 0.04
-                vals[i] = 0.001*vals[i]
-            times[i] = base_datetime
+        my_date = datetime.strptime(startdate, "%Y%m%d%H")
+        # base_datetime = my_date.strftime('%Y-%m-%dT%H:00:00Z')  # not needed
+        epoch_time = np.int64(get_epoch_time(my_date))
+        print(f' calculated epoch_time: {epoch_time}')
+
+        # vals[vals >= 0.0] *= 0.001      # mm to meters
+        errs[:] = 0.04                  # error in meters
```
Comment (BenjaminRuston) [inline src/land/ghcn_snod2ioda.py:174]:
```suggestion
        # errs[:] = 0.04                  # error in meters
```

File: src/land/ghcn_snod2ioda.py [hunk 4]
```diff
@@ -150,22 +165,24 @@ def assignValue(colrowValue, df400):
...
-                vals[i] = 0.001*vals[i]
-            times[i] = base_datetime
+        my_date = datetime.strptime(startdate, "%Y%m%d%H")
+        # base_datetime = my_date.strftime('%Y-%m-%dT%H:00:00Z')  # not needed
+        epoch_time = np.int64(get_epoch_time(my_date))
+        print(f' calculated epoch_time: {epoch_time}')
+
+        # vals[vals >= 0.0] *= 0.001      # mm to meters
+        # errs[:] = 0.04                  # error in meters
+        # errs[:] = 40.0
```
Comment (BenjaminRuston) [inline src/land/ghcn_snod2ioda.py:175]:
```suggestion
        errs[:] = 40.0                                   # error in mm
```

**Review by YoulongXia-NOAA — APPROVED** [review]
Look good. I approve this PR.

**Review by fabiolrdiniz — APPROVED** [review]
Thanks, @BenjaminRuston!

**Pull request merged**